frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboard.tsx

```diff
@@ -31,7 +43,11 @@ const MonitoringDashboard: React.FC<MonitoringDashboardProps> = ({ match }) => {
               title={q.title}
               namespace={namespace}
               graphType={q.chartType}
-              query={q.query({ namespace })}
+              query={
+                workloadName && workloadType
+                  ? q.query({ ns: namespace, workloadName, workloadType })
+                  : q.query({ namespace })
+              }
```

Comment:
can we have specs for this

Reply:
@invincibleJai I will add a test case for this.

Reply:
added tests for this. PTAL.